api/task_manager_test.cc

```diff
@@ -0,0 +1,107 @@
+#include "task_manager/test_task.hh"
+
+
+namespace api {
+
+namespace tmt = httpd::task_manager_test_json;
+using namespace json;
+
+void set_task_manager_test(http_context& ctx, routes& r) {
+    tmt::register_test_module.set(r, [&ctx] (std::unique_ptr<request> req) -> future<json::json_return_type> {
+        co_await ctx.tm.invoke_on_all([] (task_manager& tm) {
+            tm.make_test_module();
+        });
+        co_return json_void();
+    });
+
+    tmt::unregister_test_module.set(r, [&ctx] (std::unique_ptr<request> req) -> future<json::json_return_type> {
+        co_await ctx.tm.invoke_on_all([name = "test"] (task_manager& tm) -> future<> {
+            auto& modules = tm.get_modules();
+            auto it = modules.find(name);
+            if (it != modules.end()) {
+                co_await it->second->stop();
+            }
+        });
+        co_return json_void();
+    });
+
+    tmt::register_test_task.set(r, [&ctx] (std::unique_ptr<request> req) -> future<json::json_return_type> {
+        sharded<task_manager>& tms = ctx.tm;
+        auto it = req->query_parameters.find("task_id");
+        task_id id = it != req->query_parameters.end() ? task_id{utils::UUID{it->second}} : task_id::create_random_id();
+        it = req->query_parameters.find("keyspace");
+        std::string keyspace = it != req->query_parameters.end() ? it->second : "";
+        it = req->query_parameters.find("table");
+        std::string table = it != req->query_parameters.end() ? it->second : "";
+        it = req->query_parameters.find("type");
+        std::string type = it != req->query_parameters.end() ? it->second : "";
+        it = req->query_parameters.find("entity");
+        std::string entity = it != req->query_parameters.end() ? it->second : "";
+        it = req->query_parameters.find("sequence_number");
+        int sequence_number = it != req->query_parameters.end() ? boost::lexical_cast<int>(it->second) : 0;
+        it = req->query_parameters.find("parent_id");
+        task_id parent_id = it != req->query_parameters.end() ? task_id{utils::UUID{it->second}} : task_id::create_null_id();
+        co_await tms.invoke_on(0, [id, name = "test", sequence_number, keyspace = std::move(keyspace), table = std::move(table), type = std::move(type), entity = std::move(entity), parent_id] (task_manager& tm) -> future<> {
+            auto& modules = tm.get_modules();
+            auto it = modules.find(name);
+            auto _id = id;
+            if (it != modules.end()) {
+                // TODO Deexie use make_task
+                auto task_impl_ptr = std::make_unique<test_task>(it->second, id, sequence_number, std::move(keyspace), std::move(table), std::move(type), std::move(entity));
+                auto task = seastar::make_lw_shared<task_manager::task>(std::move(task_impl_ptr));
+                task->start();
+            } else {
+                co_await seastar::coroutine::return_exception(std::runtime_error("module not found"));
```

Comment:
In second thought, since we want all users of the task_manager interface to generate the same exception,
we can provide a `module_ptr find_module(std::string_view module_name)` interface in the task_manager that will throw this exception when the module name isn't found.